### ISSUE TITLE
Modern docking system for godot

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-proposals#1267]
**Describe the project you are working on:**
A Spaceship game

**Describe the problem or limitation you are having in your project:**
customization of the layout of the editor

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
I know people are working on implementing better docking so I figured I would make this suggestion..

some things I think would also make this better would be..
- allow us to drag and drop windows by dragging their title bars over to another window
- show us an idea of where we can drag and drop with something like so..
![image](https://user-images.githubusercontent.com/2453707/88605495-ee1a5e80-d02e-11ea-8f3a-d47f6e91fd24.png)

- allow us to close each window and reopen it from editor->editor windows.
- allows us to move the script window, scene window and bottom windows

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
basically something similar to unity where each window can be docked really about anywhere

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
it would be used alot and no i don't think it can be worked around.

**Is there a reason why this should be core and not an add-on in the asset library?:**
its improves workflow
